x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -0,0 +1,295 @@
+        id="xpack.csp.gcpIntegration.setupInfoContent"
+        defaultMessage="The integration will need elevated access to run some CIS benchmark rules. Select your preferred
+    method of providing the GCP credentials this integration will use. You can follow these
+    step-by-step instructions to generate the necessary credentials."
+      />
+    </EuiText>
+  </>
+);
+
+/* NEED TO FIND THE REAL URL HERE LATER*/
+const DocsLink = (
+  <EuiText color={'subdued'} size="s">
+    <FormattedMessage
+      id="xpack.csp.gcpIntegration.docsLink"
+      defaultMessage="Read the {docs} for more details"
+      values={{
+        docs: (
+          <EuiLink href="https://cloud.google.com/docs/authentication" external>
+            documentation
+          </EuiLink>
+        ),
+      }}
+    />
+  </EuiText>
+);
+
+const CredentialFileText = i18n.translate(
+  'xpack.csp.findings.gcpIntegration.gcpInputText.credentialFileText',
+  { defaultMessage: 'Path to JSON file containing the credentials and key used to subscribe' }
+);
+const CredentialJSONText = i18n.translate(
+  'xpack.csp.findings.gcpIntegration.gcpInputText.credentialJSONText',
+  { defaultMessage: 'JSON blob containing the credentials and key used to subscribe' }
+);
+
+type GcpCredentialsType = 'credentials_file' | 'credentials_json';
+type GcpFields = Record<string, { label: string; type?: 'password' | 'text' }>;
+interface GcpInputFields {
+  fields: GcpFields;
+}
+
+const gcpField: GcpInputFields = {
+  fields: {
+    project_id: {
+      label: i18n.translate('xpack.csp.gcpIntegration.projectidFieldLabel', {
+        defaultMessage: 'Project ID',
+      }),
+      type: 'text',
+    },
+    credentials_file: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsFileFieldLabel', {
+        defaultMessage: 'Credentials File',
+      }),
+      type: 'text',
+    },
+    credentials_json: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsJSONFieldLabel', {
+        defaultMessage: 'Credentials JSON',
+      }),
+      type: 'text',
+    },
+  },
+};
+
+const credentialOptionsList = [
+  {
+    label: i18n.translate('xpack.csp.gcpIntegration.credentialsFileOption', {
+      defaultMessage: 'Credentials File',
+    }),
+    text: 'Credentials File',
+  },
+  {
+    label: i18n.translate('xpack.csp.gcpIntegration.credentialsjsonOption', {
+      defaultMessage: 'Credentials JSON',
+    }),
+    text: 'Credentials JSON',
+  },
+];
+
+const getSetupFormatOptions = (): Array<{
+  id: SetupFormatGCP;
+  label: string;
+  disabled: boolean;
+}> => [
+  {
+    id: 'google_cloud_shell',
+    label: i18n.translate('xpack.csp.gcpIntegration.setupFormatOptions.googleCloudShell', {
+      defaultMessage: 'Google Cloud Shell',
+    }),
+    disabled: true,
+  },
+  {
+    id: 'manual',
+    label: i18n.translate('xpack.csp.gcpIntegration.setupFormatOptions.manual', {
+      defaultMessage: 'Manual',
+    }),
+    disabled: false,
+  },
+];
+
+interface Props {
```

Comment:
regarding my previous comment, i saw that it was resolved but without changes, what are your thoughts? you prefer to keep it as `Props`?

Reply:
oh woops I must have misunderstood your previous comments
hmmm, I was thinking on just keeping it as Props considering we don't really use it anywhere else other than this component and also I saw we use Props as the interface name for awsCredentialForm component (both this and that has the same number of parameter)
that being said, I think theres no harm on making the name more specific, I'll add it to my change